### PR TITLE
fix: covidcast_epidata

### LINKED ISSUE #53: Use consistent naming for `covidcast` `data_source`s/`source`s
Use the same base name for data-source-associated:
- parameters to `covidcast` and related functions
- elements of `covidcast_epidata()`
- columns of `covidcast` results
- etc.

Either `source` and `sources` or `data_source` and `data_sources`.

### LINKED ISSUE #64: Don't duplicate source name in `names(covidcast_epidata()$signals)`
``` r
library(epidatr)
head(names(covidcast_epidata()$signals))
#> [1] "chng.chng:smoothed_outpatient_cli"      
#> [2] "chng.chng:smoothed_adj_outpatient_cli"  
#> [3] "chng.chng:smoothed_outpatient_covid"    
#> [4] "chng.chng:smoothed_adj_outpatient_covid"
#> [5] "chng.chng:smoothed_outpatient_flu"      
#> [6] "chng.chng:smoothed_adj_outpatient_flu"
```

<sup>Created on 2023-01-20 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

This probably just requires adding an `unname` before `c`-ing the per-source signal lists together:
```r
    all_signals <- do.call(c, lapply(sources, function(x) {
        l <- c(x$signals)
        names(l) <- paste(x$source, names(l), sep = ":")
        l
    }))
```
->
```r
    all_signals <- do.call(c, unname(lapply(sources, function(x) {
        l <- c(x$signals)
        names(l) <- paste(x$source, names(l), sep = ":")
        l
    })))
```
+ updating docs, tests, etc.

Marking this P0, as fixing it is a breaking change for the entire `covidcast_epidata()$signals` interface.

### LINKED ISSUE #92: Update issues&lags exclusivity checks to also consider `as_of`
I believe `covidcast` no longer supports `as_of` together with `issues` or `lags`.  Other endpoints might still support.  We should update the client-side checks.  I believe it's also in an Issue somewhere to detect this on the server side as well.

### LINKED ISSUE #98: covid_hosp_facility_lookup should warn/error for not implemented cases
Unsure if this exists as an issue already, but the call 
```
covid_hosp_facility_lookup(state = "WY", city = "New York") %>% fetch_tbl()
```
silently just returns the state result. As more than one input is invalid, this should probably return an error/warning to that effect.

### LINKED ISSUE #109: Eliminate remaining references to "character string"
We should use, e.g., "character vector", "string" (length-1 non-NA character vector, in `rlang` terminology), or some other more familiar term.

### LINKED ISSUE #124: Address build-args typo in ci
E.g., [here](https://github.com/cmu-delphi/epidatr/actions/runs/5226403357/jobs/9437033289?pr=123).  "Warning: Unexpected input(s) 'build-args', valid inputs are ['args', 'build_args', 'error-on', 'check-dir', 'working-directory', 'upload-snapshots', 'upload-results']"

### LINKED ISSUE #145: Mark `covidcast_epidata()` as experimental

### PR DESCRIPTION
**Pull request by dshemetov**
Fix #64 
Fix #145 
Fix #53 
Fix #124 
Fix #109 
Fix #92 
Fix #98

### CODE REVIEW

File: .bumpversion.cfg
Comment (dsweber2) [inline .bumpversion.cfg:1]: I would maybe call removing a function a minor release rather than a bugfix, but nbd really

**Review by dsweber2 — APPROVED** [review]
not sure how this fixes #64? otherwise seems good!

File: .bumpversion.cfg
Comment (dshemetov) [reply .bumpversion.cfg:1]: fair

**Comment by dshemetov** [comment]
There's a sneaky `unname` that fixed #64. The code that does the parsing is pretty hard to understand tbh.

File: R/covidcast.R [hunk 1]
```diff
@@ -130,11 +121,13 @@ covidcast_epidata <- function(base_url = global_base_url, timeout_seconds = 30)
   sources <- do.call(c, lapply(response_content, parse_source, base_url = base_url))
   class(sources) <- c("covidcast_data_source_list", class(sources))
 
-  all_signals <- do.call(c, lapply(sources, function(x) {
-    l <- c(x$signals)
-    names(l) <- paste(x$source, names(l), sep = ":")
-    l
-  }))
+  all_signals <- do.call(c, unname(
```
Comment (dshemetov) [inline R/covidcast.R:124]: This is the #64 fix

**Pull request merged**